sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py

```diff
@@ -166,24 +166,39 @@ def __init__(self, **kwargs) -> None:
 
     @classmethod
     def _from_generated(
-        cls, response: "HttpResponse", deserialized_operation: "FullBackupOperation", response_headers: "Dict"
+        cls, response: HttpResponse, deserialized_operation: FullBackupOperation, response_headers: Dict
     ) -> "KeyVaultBackupResult":
         return cls(folder_url=deserialized_operation.azure_storage_blob_container_uri)
 
 
-class KeyVaultSetting(object):
+class KeyVaultSetting(Generic[SettingValueType]):
     """A Key Vault setting.
 
     :ivar str name: The name of the account setting.
-    :ivar str value: The value of the pool setting.
-    :ivar SettingType type: The type specifier of the value.
+    :ivar value: The value of the pool setting. This is a boolean if `type` is SettingType.BOOLEAN, and a string
+        otherwise. If `value` is provided as a string when `type` is SettingType.BOOLEAN, the value will be converted
+        into a boolean (True if `value` is "True" (case-insensitive); False otherwise).
+    :vartype value: SettingValueType
+    :ivar type: The type specifier of the value.
+    :vartype type: SettingType or None
     """
 
-    def __init__(self, **kwargs) -> None:
-        self.name = kwargs.get("name")
-        self.value = kwargs.get("value")
-        self.type = kwargs.get("type")
+    def __init__(
+        self,
+        *,
+        name: str,
+        value: SettingValueType,
+        type: Optional[SettingType] = None,
+        **kwargs,  # pylint:disable=unused-argument,redefined-builtin
+    ) -> None:
+        self.name = name
+        self.value = value
+        self.type = type
+
+        # If `value` was given as a string but the type is boolean, convert it to a bool
+        if hasattr(self.value, "lower") and self.type == SettingType.BOOLEAN:
+            self.value = cast(SettingValueType, self.value.lower() == "true")
```

Comment:
I ended up using `typing.cast` here and on line 204 to deal with mypy warnings such as:
```
azure\keyvault\administration\_models.py:204: error: Argument "value" to "KeyVaultSetting" has incompatible type "str"; expected "SettingValueType"  [arg-type]
```
I've done a _lot_ of searching online, but I'm not entirely sure why this has been coming up. The reason that makes the most sense is that a TypeVar `T` is expected to be the same actual type everywhere throughout the scope of `T`'s use.

Since KeyVaultSetting is a `Generic[SettingValueType]`, in the case where `self.value` ends up being a `bool`, I think mypy is upset that elsewhere `value` is being used as a `str`. Outside of using `cast`, I'm not sure how we get around this issue and indicate that the type can be converted at runtime